src/kOS/Function/FunctionBase.cs

```diff
@@ -4,11 +4,9 @@
 
 namespace kOS.Function
 {
-    public class FunctionBase
+    public abstract class FunctionBase
     {
-        public virtual void Execute(SharedObjects shared)
-        {
-        }
+        public abstract void Execute(SharedObjects shared);
```

Comment:
function base has always been abstract in practice. when i noticed it wasnt i fixed it. It should have been moved out of this PR :(